doc/types/store.qbk

```diff
@@ -0,0 +1 @@
    [bool]
    [
      Returns true if the database is open. 
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store.key_path `s.key_path()`]]
    [path_type const&]
    [
      Return the path to the key file. 
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store.key_size `s.key_size()`]]
    [std::size_t]
    [
      Return the key size associated with the database. 
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store.log_path `s.log_path()`]]
    [path_type const&]
    [
      Return the path to the log file. 
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store.open `s.open(dp, kp, lp, ec, args...)`]]
    [void]
    [
      Open a database. 
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store.basic_store `basic_store()`]]
    [ ]
    [
      Default constructor. 
      Copy constructor disallowed.
    ]
  ]
  [
    [[link nudb.ref.nudb__basic_store._basic_store `~basic_store()`]]
    [ ]
    [
      Destroy the database. 
    ]
  ]
]
[heading Valid Expressions - Free Functions]
[table [[Operation] [Return Type] [Description]]
  [
    [[link nudb.ref.nudb__create
    ``
    create<H, F, Args...>(
        dp,
        kp,
        lp,
        appnum,
        n,
        salt,
        lf,
        ec,
        args...
    )
    ``
    ]]
    [void]
    [
      Create a new database.
    ]
  ]
  [
    [[link nudb.ref.nudb__recover
    ``
    recover<H, F, Args...>(
        dp,
        kp,
        lp,
        appnum,
        n,
        n,
        lf,
        ec,
        args...
    )
    ``
    ]]
    [void]
    [
      Recover a database after it was closed unexpectedly.
    ]
  ]
  [
    [[link nudb.ref.nudb__rekey
    ``
    rekey<H, F, Args...>(
        dp,
        kp,
        lp,
        n,
        lf,
        ic,
        n,
        ec,
        pr
        args...
    )
    ``
    ]]
    [void]
    [
      Rebuild the database key.
    ]
  ]
  [
    [[link nudb.ref.nudb__verify
    ``
    verify<H, P>(
        v,
        dp,
        kp,
        lp,
        n,
        p,
        ec
    )
    ``
    ]]
    [void]
    [
      Rebuild the database key.
    ]
  ]
  [
    [[link nudb.ref.nudb__visit
    ``
    visit<C, P>(
        p,
        c,
        pr,
        lp,
        ec
    )
    ``
    ]]
    [void]
    [
      open database and invoke C for every key/data pair.
    ]
  ]
  [
    [[link nudb.ref.nudb__make_salt
    ``
    make_salt()
    ``
    ]]
    [std::uint64_t]
    [
      open database and invoke C for every key/data pair.
    ]
  ]
]

[heading Header]
[@../../include/nudb/basic_store.hpp `#include <nudb/basic_store.hpp>`]

[heading Models]
The simplest and most common usage of basic_store is the store class which is defined like this:
``
 #include <nudb/native_file.hpp>
 #include <nudb/xxhasher.hpp>
 using store = basic_store< xxhasher, native_file >;
``
[heading Example]
``
#include <nudb/basic_store.hpp>
nudb::store db;
nudb::error_code ec;
db.open("db.dat", "db.key", "db.log", ec);
``

[endsect]

```

Comment:
Weird `^M` happening here, and the file is missing a carriage return on the last line.